tests/test_class_new.rs

```diff
@@ -258,3 +258,41 @@ fn test_new_existing() {
         assert!(!obj5.is(&obj6));
     });
 }
+
+#[pyclass]
+struct NewReturnsPy;
+
+#[pymethods]
+impl NewReturnsPy {
+    #[new]
+    fn new(py: Python<'_>) -> PyResult<Py<NewReturnsPy>> {
+        Py::new(py, NewReturnsPy)
+    }
+}
+
+#[test]
+fn test_new_returns_py() {
+    Python::with_gil(|py| {
+        let obj = NewReturnsPy::new(py).unwrap();
+        assert!(obj.into_bound(py).is_exact_instance_of::<NewReturnsPy>());
+    })
+}
+
+#[pyclass]
+struct NewReturnsBound;
+
+#[pymethods]
+impl NewReturnsBound {
+    #[new]
+    fn new(py: Python<'_>) -> PyResult<Bound<'_, NewReturnsBound>> {
+        Bound::new(py, NewReturnsBound)
+    }
+}
+
+#[test]
+fn test_new_returns_bound() {
+    Python::with_gil(|py| {
+        let obj = NewReturnsBound::new(py).unwrap();
```

Comment:
This just calls `new`, it doesn't actually invoke the mechanism you added.

Take a look at `test_new_existing` in this file for how you can test it.

Reply:
Hmm that's weird, it definitely hits the new `From` impl indirectly in the call sequence (the macro-generated  `__pymethod___new____` calls `IntoPyCallbackOutput::convert` on the `Bound<T>` which calls `<Bound<T> as Into<_>>::into` and then `From::from`), but I guess its not direct enough for the coverage runner to detect. Thanks for the tip, I am very new to CI :)

Reply:
Oh wait, the Rust-side call obviously doesn't actually invode the `__pymethod___new____` stuff, duh! I'll fix it now